### ISSUE TITLE
NCCL timeout when using ``` lora_target_modules ``` gate_proj, up_proj, down_proj with MOE models

### ISSUE THREAD

**Issue by ved1beta** [axolotl-ai-cloud/axolotl#3149]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

training should run fine 

### Current behaviour

configs lags for a while and gives NCCL timeout  

### Steps to reproduce

run the given config 

### Config yaml

```yaml
base_model: Qwen/Qwen3-Coder-30B-A3B-Instruct
# Automatically upload checkpoint and final model to HF
# hub_model_id: username/custom_model_name

load_in_8bit: false
load_in_4bit: false
strict: false

datasets:
  - path: tatsu-lab/alpaca
    type: alpaca
dataset_prepared_path:
val_set_size: 0.05
output_dir: ./outputs/out

sequence_len: 2048
sample_packing: true
eval_sample_packing: true


adapter: lora
lora_model_dir:
lora_r: 32
lora_alpha: 64
lora_dropout: 0.05
lora_target_modules:
  - q_proj
  - k_proj
  - v_proj
  - o_proj
  - gate_proj
  - up_proj
  - down_proj

wandb_project:
wandb_entity:
wandb_watch:
wandb_name:
wandb_log_model:

gradient_accumulation_steps: 4
micro_batch_size: 1
num_epochs: 1
optimizer: adamw_torch_8bit
lr_scheduler: cosine
learning_rate: 0.0002

bf16: auto
tf32: true

gradient_checkpointing: true
activation_offloading: true
resume_from_checkpoint:
logging_steps: 1
flash_attention: true

warmup_ratio: 0.1
evals_per_epoch: 4
saves_per_epoch: 1
weight_decay: 0.0
special_tokens:

fsdp_version: 2
fsdp_config:
  offload_params: false
  state_dict_type: SHARDED_STATE_DICT
  auto_wrap_policy: TRANSFORMER_BASED_WRAP
  transformer_layer_cls_to_wrap: Qwen3MoeDecoderLayer
  reshard_after_forward: true
  cpu_ram_efficient_loading: true


# save_first_step: true  # uncomment this to validate checkpoint saving works with your config
```

### Possible solution

works if we commented out ``` gate_proj ``` ``` up_proj ``` ``` down_proj ``` some thing is wrong
in [lora.py]( https://github.com/axolotl-ai-cloud/axolotl/blob/main/src/axolotl/kernels/lora.py#L310-L340) calculating gate_proj, up_proj, down_proj  causes synchronization error some how 

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.12

### axolotl branch-commit

current 

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by maximrepidgey** [comment]
I encountered exactly the same issue with GLM-4.5-Air as it also uses MoE

**Comment by maximrepidgey** [comment]
Can #3030 solve it? If so, can it be merged?

**Comment by NanoCode012** [comment]
I missed this thread. Is this still an Issue? #3395 should possible be related bnb lora for moe models
